python/paddle/distributed/launch.py

```diff
@@ -117,33 +71,128 @@ def parse_args():
+        default=True,
+        help="Print the config or not")
+
     parser.add_argument(
-        '--log_dir',
+        "--selected_gpus",
         type=str,
-        default="mylog",
-        help='directory to put logs per process.')
+        default="0,1,2,3,4,5,6,7",
+        help="It's for gpu trainning and the trainning process will run on the selected_gpus,"
+        "each process is bound to a single GPU.")
+
+    parser.add_argument(
+        "--split_log_path", type=str, help="The splited log path.")
+
+    # positional
     parser.add_argument(
-        'entrypoint_script',
+        "training_script",
         type=str,
-        help="The entrypoint script to be launched in parallel,"
-        "followed by all the arguments for each process,"
-        "e.g. train.py --lr 0.1")
-    parser.add_argument('entrypoint_args', nargs=argparse.REMAINDER)
+        help="The full path to the single GPU training "
+        "program/script to be launched in parallel, "
+        "followed by all the arguments for the "
+        "training script")
+
+    # rest from the training program
+    parser.add_argument('training_script_args', nargs=REMAINDER)
     return parser.parse_args()
 
 
-def main():
-    args = parse_args()
+def start_procs(args):
+    """
+    """
+    procs = []
+    log_fns = []
+
+    default_env = os.environ.copy()
+
+    node_id = args.node_id
+    node_ips = [x.strip() for x in args.node_ips.split(',')]
+    assert node_id >= 0 and node_id < len(node_ips), \
+            "node_id:{} must be in range of the node_ips:{}".format(node_id, node_ips)
+
+    current_node_ip = node_ips[node_id]
+    num_nodes = len(node_ips)
+    selected_gpus = [x.strip() for x in args.selected_gpus.split(',')]
+    selected_gpus_num = len(selected_gpus)
+
+    trainers_endpoints = ""
+    for ip in node_ips:
+        for i in range(selected_gpus_num):
+            if trainers_endpoints != "":
+                trainers_endpoints += ","
+            trainers_endpoints += "%s:617%d" % (ip, i)
+
+    nranks = num_nodes * selected_gpus_num
+
+    if args.print_config:
+        print("trainers_endpoints:", trainers_endpoints, ", node_id:", node_id,
+              ", current_node_ip:", current_node_ip, ", num_nodes:", num_nodes,
+              ", node_ips:", node_ips, ", nranks:", nranks)
+
+    current_env = copy.copy(default_env)
+    procs = []
+    cmds = []
+    for i in range(0, selected_gpus_num):
+        current_env.update({
+            "FLAGS_selected_gpus": "%s" % selected_gpus[i],
+            "PADDLE_TRAINER_ID": "%d" % (node_id * selected_gpus_num + i),
+            "PADDLE_CURRENT_ENDPOINT": "%s:%d" % (current_node_ip, 6170 + i),
+            "PADDLE_TRAINERS_NUM": "%d" % nranks,
+            "PADDLE_TRAINER_ENDPOINTS": trainers_endpoints
+        })
+
+        cmd = [sys.executable, "-u", args.training_script
+               ] + args.training_script_args
+
+        cmds.append(cmd)
+
+        if args.split_log_path is not None:
```

Comment:
useful argument, but the name is difficult to follow. my first reaction is change "/split/log/path" to ["split", "log", "path"].

Reply:
the original log_dir is just OK i think.

Reply:
Done.